задания 10-ого спринта

malenkii-myk opened the pull request.

malenkii-myk committed:
задания 10-ого спринта

faritowich commented on app/src/main/java/com/practicum/playlistmaker/TracksViewHolder.kt:
```diff
@@ -0,0 +1,39 @@
+        artistName.text = model.artistName
+        trackTime.text = model.trackTime
+
+        if (!model.artworkUrl100.isNullOrEmpty()) {
```
`model.artworkUrl100.isNotEmpty()`

faritowich commented on app/src/main/java/com/practicum/playlistmaker/TracksViewHolder.kt:
```diff
@@ -0,0 +1,39 @@
+            Glide.with(itemView)
+                .load(R.drawable.track_placeholder)
+                .into(artImg)
+        }
```
Тут вообще эта логика не нужна. Если не получится загрузить обложку, то и так подставится плейсхолдер из `.placeholder(R.drawable.track_placeholder)`

malenkii-myk replied:
Моя логика такова (возможно она ошибочная) - может же не быть обложки у трека, в этом случае я перезагружаю обложку в ItemView на placeholder, т.к. я понял из курса, что эти ItemView переиспользуются и может такая сложиться ситуация что все textview обновятся новой информацией, а картинка останется от предыдущего трека.

faritowich commented on app/src/main/res/layout/track_view.xml:
```diff
@@ -0,0 +1,71 @@
+        style="@style/track_list_item_title"
+        android:id="@+id/trackName"
+        tools:text="Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum"
+        android:layout_marginStart="53dp"
```
Не стоит в таких случаях задавать взаимное расположение (в данном случае ImageView относительно TextView) большим отступом от начала экрана. Это моветон. Тут напрашивается горизонтальный LinearLayout вместо FrameLayout, где сперва идет ImageView, а потом TextView.

faritowich commented on app/src/main/res/layout/track_view.xml:
```diff
@@ -0,0 +1,71 @@
+            android:layout_width="280dp"
+            android:layout_height="13dp"
+            android:layout_marginStart="53dp"
+            android:layout_marginTop="26dp"
```
Очень плохая верстка на этом экране, надо всё переделать по хорошему. Смысл такой - если у нас идут элементы подряд, то используем LinearLayout. На этом экране получится несколько вложенных LinearLayout

malenkii-myk replied:
как раз я и думал делать через LinearLayout но побоялся лишней вложенности, думал она будет критичной. 
Понял, переделаю

faritowich commented on app/src/main/res/layout/track_view.xml:
```diff
@@ -0,0 +1,71 @@
+        android:layout_gravity="center|end"
+        android:src="@drawable/arrow_forward"
+        app:tint="@color/gray" />
+</FrameLayout>
```
Поплыло форматирование в файле. Есть хороший хоткей, можно нажать Alt + Ctrl + L и студия красиво отформатирует весь код на экране. Советую прожимать каждый раз после окончания работы в файле

malenkii-myk committed:
работа над замечаниями
+ isNotEmpty()
+ track_view переделано через LinearLayout
+ toolbar переделан через LinearLayout
